pkg/cloudprovider/types.go

```diff
@@ -373,3 +373,24 @@ func IsNodeClassNotReadyError(err error) bool {
 	var nrError *NodeClassNotReadyError
 	return errors.As(err, &nrError)
 }
+
+// InstanceCreateError is an error type returned by CloudProviders when instance creation fails
+type InstanceCreateError struct {
+	error
+	message string
+}
+
+func NewInstanceCreateError(err error, message string) *InstanceCreateError {
```

Comment:
We should handle parsing this out and setting the condition in the lifecycle controller in this same code change -- otherwise, there's not a ton of value in just adding the error